src/cdi_ndi/NDI_Base.cc

```diff
@@ -55,6 +55,80 @@ void NDI_Base::warn_ndi_version_mismatch(std::string const & /*gendir*/) {}
 // CONSTRUCTORS
 //------------------------------------------------------------------------------------------------//
 
+//============================================================================//
+// Stubbed implementation when NDI is unavailable
+//============================================================================//
+
+#ifndef NDI_FOUND
+
+//! Constructor for generic NDI reader- throws when NDI not available
+NDI_Base::NDI_Base(const std::string & /*dataset_in*/, const std::string & /*library_in*/) {
+  Insist(0, "NDI default gendir path only available when NDI is found.");
+}
+
+#else
+
+NDI_Base::NDI_Base(const std::string &dataset_in, const std::string &library_in)
+    : gendir(rtt_dsxx::get_env_val<std::string>("NDI_GENDIR_PATH").second), dataset(dataset_in),
+      library(library_in) {
+
+  Require(rtt_dsxx::fileExists(gendir));
+
+  Require(gendir.length() > 0);
+  Require(dataset.length() > 0);
+  Require(library.length() > 0);
+  warn_ndi_version_mismatch(gendir);
+}
+
+#endif
+
+//============================================================================//
+// Stubbed implementation when NDI is unavailable
+//============================================================================//
+
+#ifndef NDI_FOUND
+
+//! Constructor for generic NDI reader- throws when NDI not available
+NDI_Base::NDI_Base(const std::string & /*gendir_in*/, const std::string & /*dataset_in*/,
+                   const std::string & /*library_in*/) {
+  Insist(0, "NDI default gendir path only available when NDI is found.");
+}
+
+#else
+
+NDI_Base::NDI_Base(const std::string &gendir_in, const std::string &dataset_in,
+                   const std::string &library_in)
+    : gendir(gendir_in), dataset(dataset_in), library(library_in) {
+
+  Require(rtt_dsxx::fileExists(gendir));
+
+  Require(gendir.length() > 0);
+  Require(dataset.length() > 0);
+  Require(library.length() > 0);
+  warn_ndi_version_mismatch(gendir);
+}
+
+#endif
+
+//================================================================================================//
+// Stubbed implementation when NDI is unavailable
+//================================================================================================//
+
+#ifndef NDI_FOUND
+
+//! Constructor for generic NDI reader- throws when NDI not available
+NDI_Base::NDI_Base(const std::string & /*gendir_in*/, const std::string & /*dataset_in*/,
+                   const std::string & /*library_in*/, const std::string & /*reaction_in*/,
+                   const std::vector<double> /*mg_e_bounds_in*/) {
```

Comment:
These `reaction` and `mg_e_bounds` fields should probably be moved up from `NDI_Base` since they aren't used by `NDI_CP_Eloss`

Reply:
This would also cut down on the now fairly large number of constructors for `NDI_Base`

Reply:
that seems reasonable to me! :)

Reply:
Done